backend/src/v2/driver/driver.go

```diff
@@ -647,23 +686,52 @@ func extendPodSpecPatch(
 					},
 				},
 			}
-			secretEnvVar.ValueFrom.SecretKeyRef.LocalObjectReference.Name = secretAsEnv.GetSecretName()
+
+			var secretName string
+			if secretAsEnv.SecretNameParameter != nil {
+				resolvedSecretName, err := resolveK8sParameter(ctx, opts, dag, pipeline, mlmd,
+					secretAsEnv.SecretNameParameter, inputParams)
+				if err != nil {
+					return fmt.Errorf("failed to resolve secret name: %w", err)
+				}
+				secretName = resolvedSecretName.GetStringValue()
+			} else {
+				secretName = secretAsEnv.SecretName
```

Comment:
Similar to above, is it possible that both are unset based on the Protobuf definition?